docs/proposals/scheduling/20240131-pod-level-numa-policy.md

```diff
@@ -0,0 +1,320 @@
+same numa. This means to split the nodes in cluster into static groups. In large clusters, with lots of nodes that can serve 
+different purposes, it's not unreasonable to dedicate a node (or set of nodes) to a certain numa topology policy, and 
+direct pods to those nodes using a nodeSelector. It's mostly problematic in smaller clusters where you don't have the 
+luxury of special-purposing nodes like this and you want to binpack pods onto nodes as tightly as possible (while still 
+reaping the benefits of topology-alignment). So we need a way to specify numa topology policy for each pod.
+
+## Proposal
+
+### Goal
+
+- Allow users to specify numa topology policy on workloads. 
+- Protect the QoS requirement for workloads with `SingleNUMANode` policy by preventing pods that cross
+numa and pods with `SingleNUMANode` policy from being deployed on the same numa.
+- New API should be compatible with numa topology policy on nodes. This means if users don't specify NUMA scheduling 
+policy on workloads, numa topology policy on nodes should still work.
+
+### Non-Goal
+
+- Change the device and CPU, Memory allocation rules during NUMA-aware scheduling.
+
+### Use Cases
+
+- As a cluster user, I don't have privileges to set the numa topology policy on the node, and I want to set `SingleNUMANode` 
+for my pods so that they can be binpacked as tightly as possible.
+- As a cluster admin, split my cluster into 2 small cluster according to numa topology policy may reduce gpu utilization.
+- As a cluster user, I hope there is a method for me to indicate that my workload do not want to be placed with a cross
+numa workload because these workloads may use too much memory and result in my inability to obtain the requested memory 
+resources.
+
+[//]: # (- As a cluster admin, I hope there is no inference between jobs with different numa topology policy. This option should be a global )
+
+[//]: # (setting so that I can enable and disable it easily.)
+
+### API
+
+We will introduce a new property in pod `scheduling.koordinator.sh/resource-spec`: `numaTopologyPolicy`. 
+The value of this label can be `""`, `Restricted`, `SingleNUMANode`, `BestEffort`. The default value of this label is `""`. 
+The meaning of these values is same with the value of policy on node.
+
+#### SingleNUMANodeExclusive
+
+To protect the QoS for the pod with `SingleNUMANode` policy, pods that cross numa should not be placed on the same numa 
+with the pod being scheduled on one numa. Sometimes this may lead to situations where workloads that use multiple NUMAs 
+could never be scheduled. Therefore, we allow some critical workloads to break this restriction.
+
+We will introduce a new property in pod `scheduling.koordinator.sh/resource-spec`: 
+`singleNUMANodeExclusive` to reach the goal, the value of this property can be:
+- `preferred`: a numa with a SingleNUMANode pod will not be scheduled another pod with multi-numa if there is another idle numa.
```

Comment:
Keep `preferred` and `required` in the same style as other fields such as `numaTopologyPolicy`, either start with uppercase or lowercase.